Assignment5/carlomagnani_21A285/Assignment5_CarloMagnani_final.py

```diff
@@ -0,0 +1,297 @@
+
+# SPARQL:
+
+#select distinct ?award{
+#  wd:Q2685 wdt:P166 ?award
+#}LIMIT 100
+
+# Kypher:
+kgtk("""
+    query -i all \
+    --match '(:Q2685)-[:P166]->(awards)' \
+    --return 'count(awards) as Number_of_awards'
+""")
+
+
+# Retrieve at least two members of Schwarzenegger's political party. Make sure only persons are returned
+# SPARQL:
+
+#select ?politician 
+#where{
+#    wd:Q2685 wdt:P102 ?party.
+#    ?politician wdt:P102 ?party.
+#    ?politician wdt:P31  wd:Q5      
+#}LIMIT 2
+
+# Kypher:
+kgtk("""
+    query -i all \
+    --match '(:Q2685)-[:P102]->(party)<-[:P102]-(politician)-[:P31]->(:Q5)' \
+    --return 'politician as politician'
+    --limit 2
+    / add-labels
+""")
+
+
+# What are the properties that describe an artist?
+
+# In theory this one is heavy on Wikidata
+
+# SPARQL: 
+
+#select distinct ?prop 
+#where{
+# ?artist wdt:P106 wd:Q483501.
+# ?artist ?prop ?values
+#}
+
+
+# In Kypher:
+kgtk("""
+    query -i all \
+    --match '(:Q483501)<-[:P31]-(artist)-[p]->()' \
+    --return 'distinct p.label as Property'
+    / add-labels
+""")
+
+
+# And a film director?
+# SPARQL: 
+
+#select distinct ?prop 
+#where{
+# ?director wdt:P106 wd:Q2526255.
+# ?director ?prop ?values
+#}
+
+# In Kypher:
+kgtk("""
+    query -i all \
+    --match '(:Q2526255)<-[:P106]-(director)-[p]->()' \
+    --return 'distinct p.label as Property'
+    / add-labels
+""")
+
+
+# Embeddings. Run the noebook https://colab.research.google.com/drive/1A55l10voA4jnjoju3fojJWY3buLfaR4i?usp=sharing. 
+# Which are the top 10 similar entities to Schwarzenegger? (list below) 
+
+#1) Arnold Schwarzenegger (himeslf); 2) Hugh O'Brian; 3) John Larroquette; 4) Carl Reiner; 5) Harvey Fierstein; 6) Tom Sizemore; 7) Randy Quaid; 8) Gene Kelly; 9) DeForest Kelley; 10) Robert Stack
+
+#########################################################################
+#########################################################################
+
+# ## Network analysis
+# Print all the paths between Schwarzenegger and Trump
+# 
+# ## Note that **you have to create a file `paths.tsv` with the node pairs you want to find the paths for. Upload it in the "content" folder**
+# 
+
+
+#%%bash
+#cat <<EOF >$TEMP/path-query.tsv
+#node1	node2	label
+#Q2685	Q22686	path
+
+
+kgtk("""
+    add-labels -i $TEMP/path-query.tsv
+""")
+
+
+# Calculate all the paths between Trump and Schwarzenegger (max hops: 3)
+# TO DO    
+kgtk("""
+    paths -i all
+        --verbose False
+        --statistics-only True
+        --path-file $TEMP/path-query.tsv
+        --max_hops 3
+    -o $TEMP/path-results.tsv
+""")
+
+#!head $TEMP/path-results.tsv
+
+
+# Retrieve all the family of Schwarzenegger (child/father/mother/sibling/spouse relationships)
+
+kgtk("""
+    query -i all \
+        --match '(:Q2685)-[p]->(rel)' \
+        --where 'p.label in ["P22","P25","P3373","P26","P40"]' \
+        --return 'p.label as Relationship, rel as Relative'
+    / add-labels
+""")
```

Comment:
Not correct, this should be done with the reachable nodes function. This query does not reach many reslationships.